### PR TITLE
use EXPENSIVE_CHECKS macro to enable  MF.verify from PPCMIPeephole pass

### PR DESCRIPTION
**Pull request by diggerlin**
verifying the machine instructions of the functions in the PPCMIPeephole pass will cause a huge compile time increase in some cases. use EXPENSIVE_CHECKS macro to enable  MF.verify from PPCMIPeephole pass.

### CODE REVIEW

File: llvm/lib/Target/PowerPC/PPCMIPeephole.cpp [hunk 1]
```diff
@@ -177,11 +177,6 @@ struct PPCMIPeephole : public MachineFunctionPass {
     if (skipFunction(MF.getFunction()))
       return false;
     bool Changed = simplifyCode();
-#ifndef NDEBUG
```
Comment (RolandF77) [inline llvm/lib/Target/PowerPC/PPCMIPeephole.cpp:180]: There is a comment at line 19 saying that the verifier will be run. It should be removed if the verifier will not be run.

Comment (RolandF77) [reply llvm/lib/Target/PowerPC/PPCMIPeephole.cpp:180]: Could we put this under EXPENSIVE_CHECKS rather than removing it?

Comment (chenzheng1030) [reply llvm/lib/Target/PowerPC/PPCMIPeephole.cpp:180]: Totally removing this sounds not good to save compile time, functionality issue will be hidden. We have run machine verifier many times, why the one here matters?

Comment (diggerlin) [reply llvm/lib/Target/PowerPC/PPCMIPeephole.cpp:180]: this `MF.verify`  cause Compile time regression with 502.gcc at BASE and PEAK from 62 seconds to 255 seconds

Comment (nikic) [reply llvm/lib/Target/PowerPC/PPCMIPeephole.cpp:180]: Isn't the verifier run after every pass anyway if EXPENSIVE_CHECKS are enabled?

Comment (diggerlin) [reply llvm/lib/Target/PowerPC/PPCMIPeephole.cpp:180]: I do not think so ,  the verifier only run after the passes  which has 

```
#if defined(EXPENSIVE_CHECKS)
  MF.verify();
#endif
```

Comment (RolandF77) [reply llvm/lib/Target/PowerPC/PPCMIPeephole.cpp:180]: > Isn't the verifier run after every pass anyway if EXPENSIVE_CHECKS are enabled?

Tried and can confirm that this is the case.

**Comment by diggerlin** [comment]
since using ` EXPENSIVE_CHECKS` can cause a extremely long testing time for llvm-lit , I closed the issue, We will find a better solution later on.

**Pull request closed without merging**